ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx

```diff
@@ -150,97 +145,105 @@ function NetworkGraphPage() {
+                        activeNodes,
+                        policyNodeMap,
+                        namespacesFromUrl
+                    );
 
-                        // create extraneous flows graph
-                        const extraneousFlowsDataModel = createExtraneousFlowsModel(
-                            policyDataModel,
-                            activeNodeMap,
-                            activeEdgeMap,
-                            namespacesFromUrl
-                        );
+                    // create extraneous flows graph
+                    const extraneousFlowsDataModel = createExtraneousFlowsModel(
+                        policyDataModel,
+                        activeNodeMap,
+                        activeEdgeMap,
+                        namespacesFromUrl
+                    );
 
-                        const newUpdatedTimestamp = new Date();
-                        // show only hours and minutes, use options with the default locale - use an empty array
-                        const lastUpdatedDisplayTime = newUpdatedTimestamp.toLocaleTimeString([], {
-                            hour: 'numeric',
-                            minute: '2-digit',
-                        });
-                        setLastUpdatedTime(lastUpdatedDisplayTime);
-                        setPrevEpochCount(epoch);
-                        setCurrentEpochCount(epoch);
+                    const newUpdatedTimestamp = new Date();
+                    // show only hours and minutes, use options with the default locale - use an empty array
+                    const lastUpdatedDisplayTime = newUpdatedTimestamp.toLocaleTimeString([], {
+                        hour: 'numeric',
+                        minute: '2-digit',
+                    });
+                    setLastUpdatedTime(lastUpdatedDisplayTime);
 
-                        setModels({
-                            activeModel: activeDataModel,
-                            extraneousModel: extraneousFlowsDataModel,
-                        });
-                    })
-                    .catch(() => {
-                        // TODO
-                    })
-                    .finally(() => setIsLoading(false));
-            }
+                    // Set the epoch to the most recent value from the server, since the state should now be up to date
+                    // with that value at worst.
+                    setPrevEpochCount(currentEpochCount);
+
+                    setModels({
+                        activeModel: activeDataModel,
+                        extraneousModel: extraneousFlowsDataModel,
+                    });
+                })
+                .catch(() => {
+                    // TODO
+                })
+                .finally(() => setIsLoading(false));
         }
+    }
+
+    // Epoch counts are tracked separately for each cluster, so reset them when the cluster changes.
+    useEffect(() => {
```

Comment:
Without this, if a user changes cluster to one with an epoch less than their previous cluster, they will never get notified of cluster updates.